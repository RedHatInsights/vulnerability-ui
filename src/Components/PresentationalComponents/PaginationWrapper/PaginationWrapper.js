import { Pagination, PaginationVariant } from '@patternfly/react-core';
import TableToolbar from '@redhat-cloud-services/frontend-components/TableToolbar';

import propTypes from 'prop-types';
import React from 'react';
import { DEFAULT_PAGE_SIZE } from '../../../Helpers/constants';

const PaginationWrapper = ({ apply, meta, variant }) => {
    const { page, total_items: totalItems, page_size: pageSize } = meta;

    const handleChangePage = (_event, pageNumber) => apply({ page: pageNumber });
    const handleSetPageSize = (_event, perPage) => apply({ page_size: perPage, page: 1 });

    return (
        <TableToolbar isFooter>
            <Pagination
                page={page || 1}
                itemCount={totalItems || 0}
                perPage={pageSize || DEFAULT_PAGE_SIZE}
                onSetPage={handleChangePage}
                onPerPageSelect={handleSetPageSize}
                variant={PaginationVariant[variant]}
                ouiaId={'pagination-'.concat(PaginationVariant[variant])}
                isDisabled={!totalItems}
            />
        </TableToolbar>
    );
};

PaginationWrapper.defaultProps = {
    variant: 'bottom'
};

PaginationWrapper.propTypes = {
    apply: propTypes.func.isRequired,
    meta: propTypes.object.isRequired,
    variant: propTypes.string
};

export default PaginationWrapper;
