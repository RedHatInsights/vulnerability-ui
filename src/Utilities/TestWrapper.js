import propTypes from 'prop-types';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import { IntlProvider } from '@redhat-cloud-services/frontend-components-translations';

const TestWrapper = ({ children, store }) => {
    return (
        <IntlProvider locale="en">
            { store ?
                <Provider store={store}>
                    <Router>
                        {children}
                    </Router>
                </Provider> : children
            }
        </IntlProvider>
    );
};

TestWrapper.propTypes = {
    children: propTypes.node,
    store: propTypes.object
};

export default TestWrapper;
