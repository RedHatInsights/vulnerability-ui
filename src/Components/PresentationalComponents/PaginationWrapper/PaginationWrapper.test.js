import PaginationWrapper from './PaginationWrapper'
import { render, screen } from '@testing-library/react';
import TestWrapper from '../../../Utilities/TestWrapper';
import userEvent from '@testing-library/user-event';

describe('PaginationWrapper', () => {
    const meta = {
        page: 1,
        page_size: 20,
        pages: 6,
        total_items: 138
    };

    const applyMock = jest.fn(filters => { return filters });

    it('Should render with default params', () => {
        const defaultMeta = { pages: 6 }
        const { asFragment } = render(
            <TestWrapper>
                <PaginationWrapper variant="bottom" meta={defaultMeta} apply={applyMock} />
            </TestWrapper>
        );

        expect(asFragment()).toMatchSnapshot();
    });

    it('Should change page number', () => {
        render(
            <TestWrapper>
                <PaginationWrapper variant="bottom" meta={meta} apply={applyMock} />
            </TestWrapper>
        );

        userEvent.click(screen.getByLabelText('Go to next page'));
        expect(applyMock).toHaveReturnedWith({
            'page': meta.page + 1
        });
    });

    it('Should change page size number', () => {
        render(
            <TestWrapper>
                <PaginationWrapper variant="bottom" meta={meta} apply={applyMock} />
            </TestWrapper>
        );

        userEvent.click(screen.getByLabelText('Items per page'));
        userEvent.click(screen.getByRole('menuitem', { name: '10 per page' }));

        expect(applyMock).toHaveReturnedWith({
            'page_size': 10,
            'page': 1
        });
    });
});
