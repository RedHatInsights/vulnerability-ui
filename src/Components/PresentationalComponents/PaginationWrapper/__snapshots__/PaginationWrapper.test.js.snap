// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationWrapper Should render with default params 1`] = `
<PaginationWrapper
  apply={[MockFunction]}
  meta={
    Object {
      "pages": 6,
    }
  }
  variant="bottom"
>
  <Pagination
    className=""
    defaultToFullPage={false}
    firstPage={1}
    isCompact={false}
    isDisabled={false}
    isSticky={false}
    itemCount={0}
    itemsEnd={null}
    itemsStart={null}
    offset={0}
    onFirstClick={[Function]}
    onLastClick={[Function]}
    onNextClick={[Function]}
    onPageInput={[Function]}
    onPerPageSelect={[Function]}
    onPreviousClick={[Function]}
    onSetPage={[Function]}
    ouiaId="pagination-bottom"
    ouiaSafe={true}
    page={1}
    perPage={20}
    perPageOptions={
      Array [
        Object {
          "title": "10",
          "value": 10,
        },
        Object {
          "title": "20",
          "value": 20,
        },
        Object {
          "title": "50",
          "value": 50,
        },
        Object {
          "title": "100",
          "value": 100,
        },
      ]
    }
    titles={
      Object {
        "currPage": "Current page",
        "items": "",
        "itemsPerPage": "Items per page",
        "optionsToggle": "Items per page",
        "page": "",
        "paginationTitle": "Pagination",
        "perPageSuffix": "per page",
        "toFirstPage": "Go to first page",
        "toLastPage": "Go to last page",
        "toNextPage": "Go to next page",
        "toPreviousPage": "Go to previous page",
      }
    }
    toggleTemplate={[Function]}
    variant="bottom"
    widgetId="pagination-options-menu"
  >
    <div
      className="pf-c-pagination pf-m-bottom"
      data-ouia-component-id="pagination-bottom"
      data-ouia-component-type="PF4/Pagination"
      data-ouia-safe={true}
      id="pagination-options-menu-0"
    >
      <PaginationOptionsMenu
        className=""
        defaultToFullPage={false}
        dropDirection="up"
        firstIndex={0}
        isDisabled={false}
        itemCount={0}
        itemsPerPageTitle="Items per page"
        itemsTitle=""
        lastIndex={0}
        lastPage={0}
        onPerPageSelect={[Function]}
        optionsToggle="Items per page"
        page={0}
        perPage={20}
        perPageOptions={
          Array [
            Object {
              "title": "10",
              "value": 10,
            },
            Object {
              "title": "20",
              "value": 20,
            },
            Object {
              "title": "50",
              "value": 50,
            },
            Object {
              "title": "100",
              "value": 100,
            },
          ]
        }
        perPageSuffix="per page"
        toggleTemplate={[Function]}
        widgetId="pagination-options-menu"
      >
        <DropdownWithContext
          autoFocus={true}
          className=""
          direction="up"
          dropdownItems={
            Array [
              <DropdownItem
                className=""
                component="button"
                data-action="per-page-10"
                onClick={[Function]}
              >
                10
                 per page
              </DropdownItem>,
              <DropdownItem
                className="pf-m-selected"
                component="button"
                data-action="per-page-20"
                onClick={[Function]}
              >
                20
                 per page
                <div
                  className="pf-c-options-menu__menu-item-icon"
                >
                  <CheckIcon
                    color="currentColor"
                    noVerticalAlign={false}
                    size="sm"
                  />
                </div>
              </DropdownItem>,
              <DropdownItem
                className=""
                component="button"
                data-action="per-page-50"
                onClick={[Function]}
              >
                50
                 per page
              </DropdownItem>,
              <DropdownItem
                className=""
                component="button"
                data-action="per-page-100"
                onClick={[Function]}
              >
                100
                 per page
              </DropdownItem>,
            ]
          }
          isGrouped={false}
          isOpen={false}
          isPlain={true}
          menuAppendTo="inline"
          onSelect={[Function]}
          position="left"
          toggle={
            <OptionsToggle
              firstIndex={0}
              isDisabled={false}
              isOpen={false}
              itemCount={0}
              itemsPerPageTitle="Items per page"
              itemsTitle=""
              lastIndex={0}
              onToggle={[Function]}
              optionsToggle="Items per page"
              parentRef={null}
              showToggle={true}
              toggleTemplate={[Function]}
              widgetId="pagination-options-menu"
            />
          }
        >
          <div
            className="pf-c-options-menu pf-m-top"
            data-ouia-component-type="PF4/PaginationOptionsMenu"
            data-ouia-safe={true}
          >
            <OptionsToggle
              aria-haspopup={true}
              firstIndex={0}
              getMenuRef={[Function]}
              id="pf-dropdown-toggle-id-0"
              isDisabled={false}
              isOpen={false}
              isPlain={true}
              itemCount={0}
              itemsPerPageTitle="Items per page"
              itemsTitle=""
              key=".0"
              lastIndex={0}
              onEnter={[Function]}
              onToggle={[Function]}
              optionsToggle="Items per page"
              parentRef={
                Object {
                  "current": <div
                    class="pf-c-options-menu pf-m-top"
                    data-ouia-component-type="PF4/PaginationOptionsMenu"
                    data-ouia-safe="true"
                  >
                    <div
                      class="pf-c-options-menu__toggle pf-m-plain pf-m-text"
                    >
                      <span
                        class="pf-c-options-menu__toggle-text"
                      >
                        <b>
                          0
                           - 
                          0
                        </b>
                         
                        of 
                        <b>
                          0
                        </b>
                         
                        
                      </span>
                      <button
                        aria-expanded="false"
                        aria-label="Items per page"
                        class="  pf-c-options-menu__toggle-button"
                        data-ouia-component-id="OUIA-Generated-DropdownToggle-1"
                        data-ouia-component-type="PF4/DropdownToggle"
                        data-ouia-safe="true"
                        disabled=""
                        id="pagination-options-menu-toggle-0"
                        type="button"
                      >
                        <span
                          class="pf-c-options-menu__toggle-button-icon"
                        >
                          <svg
                            aria-hidden="true"
                            fill="currentColor"
                            height="1em"
                            role="img"
                            style="vertical-align: -0.125em;"
                            viewBox="0 0 320 512"
                            width="1em"
                          >
                            <path
                              d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                            />
                          </svg>
                        </span>
                      </button>
                    </div>
                  </div>,
                }
              }
              showToggle={true}
              toggleTemplate={[Function]}
              widgetId="pagination-options-menu"
            >
              <div
                className="pf-c-options-menu__toggle pf-m-plain pf-m-text"
              >
                <span
                  className="pf-c-options-menu__toggle-text"
                >
                  <ToggleTemplate
                    firstIndex={0}
                    itemCount={0}
                    itemsTitle=""
                    lastIndex={0}
                  >
                    <b>
                      0
                       - 
                      0
                    </b>
                     
                    of 
                    <b>
                      0
                    </b>
                     
                  </ToggleTemplate>
                </span>
                <DropdownToggle
                  aria-label="Items per page"
                  className="pf-c-options-menu__toggle-button"
                  id="pagination-options-menu-toggle-0"
                  isDisabled={true}
                  isOpen={false}
                  onEnter={[Function]}
                  onToggle={[Function]}
                  parentRef={
                    Object {
                      "current": <div
                        class="pf-c-options-menu pf-m-top"
                        data-ouia-component-type="PF4/PaginationOptionsMenu"
                        data-ouia-safe="true"
                      >
                        <div
                          class="pf-c-options-menu__toggle pf-m-plain pf-m-text"
                        >
                          <span
                            class="pf-c-options-menu__toggle-text"
                          >
                            <b>
                              0
                               - 
                              0
                            </b>
                             
                            of 
                            <b>
                              0
                            </b>
                             
                            
                          </span>
                          <button
                            aria-expanded="false"
                            aria-label="Items per page"
                            class="  pf-c-options-menu__toggle-button"
                            data-ouia-component-id="OUIA-Generated-DropdownToggle-1"
                            data-ouia-component-type="PF4/DropdownToggle"
                            data-ouia-safe="true"
                            disabled=""
                            id="pagination-options-menu-toggle-0"
                            type="button"
                          >
                            <span
                              class="pf-c-options-menu__toggle-button-icon"
                            >
                              <svg
                                aria-hidden="true"
                                fill="currentColor"
                                height="1em"
                                role="img"
                                style="vertical-align: -0.125em;"
                                viewBox="0 0 320 512"
                                width="1em"
                              >
                                <path
                                  d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                                />
                              </svg>
                            </span>
                          </button>
                        </div>
                      </div>,
                    }
                  }
                >
                  <Toggle
                    aria-label="Items per page"
                    bubbleEvent={false}
                    className="pf-c-options-menu__toggle-button"
                    data-ouia-component-id="OUIA-Generated-DropdownToggle-1"
                    data-ouia-component-type="PF4/DropdownToggle"
                    data-ouia-safe={true}
                    getMenuRef={null}
                    id="pagination-options-menu-toggle-0"
                    isActive={false}
                    isDisabled={true}
                    isOpen={false}
                    isPlain={false}
                    isPrimary={false}
                    isSplitButton={false}
                    onEnter={[Function]}
                    onToggle={[Function]}
                    parentRef={
                      Object {
                        "current": <div
                          class="pf-c-options-menu pf-m-top"
                          data-ouia-component-type="PF4/PaginationOptionsMenu"
                          data-ouia-safe="true"
                        >
                          <div
                            class="pf-c-options-menu__toggle pf-m-plain pf-m-text"
                          >
                            <span
                              class="pf-c-options-menu__toggle-text"
                            >
                              <b>
                                0
                                 - 
                                0
                              </b>
                               
                              of 
                              <b>
                                0
                              </b>
                               
                              
                            </span>
                            <button
                              aria-expanded="false"
                              aria-label="Items per page"
                              class="  pf-c-options-menu__toggle-button"
                              data-ouia-component-id="OUIA-Generated-DropdownToggle-1"
                              data-ouia-component-type="PF4/DropdownToggle"
                              data-ouia-safe="true"
                              disabled=""
                              id="pagination-options-menu-toggle-0"
                              type="button"
                            >
                              <span
                                class="pf-c-options-menu__toggle-button-icon"
                              >
                                <svg
                                  aria-hidden="true"
                                  fill="currentColor"
                                  height="1em"
                                  role="img"
                                  style="vertical-align: -0.125em;"
                                  viewBox="0 0 320 512"
                                  width="1em"
                                >
                                  <path
                                    d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                                  />
                                </svg>
                              </span>
                            </button>
                          </div>
                        </div>,
                      }
                    }
                  >
                    <button
                      aria-expanded={false}
                      aria-label="Items per page"
                      className="  pf-c-options-menu__toggle-button"
                      data-ouia-component-id="OUIA-Generated-DropdownToggle-1"
                      data-ouia-component-type="PF4/DropdownToggle"
                      data-ouia-safe={true}
                      disabled={true}
                      id="pagination-options-menu-toggle-0"
                      onClick={[Function]}
                      onKeyDown={[Function]}
                      type="button"
                    >
                      <span
                        className="pf-c-options-menu__toggle-button-icon"
                      >
                        <CaretDownIcon
                          color="currentColor"
                          noVerticalAlign={false}
                          size="sm"
                        >
                          <svg
                            aria-hidden={true}
                            aria-labelledby={null}
                            fill="currentColor"
                            height="1em"
                            role="img"
                            style={
                              Object {
                                "verticalAlign": "-0.125em",
                              }
                            }
                            viewBox="0 0 320 512"
                            width="1em"
                          >
                            <path
                              d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                            />
                          </svg>
                        </CaretDownIcon>
                      </span>
                    </button>
                  </Toggle>
                </DropdownToggle>
              </div>
            </OptionsToggle>
          </div>
        </DropdownWithContext>
      </PaginationOptionsMenu>
      <Navigation
        className=""
        currPage="Current page"
        firstPage={1}
        isCompact={false}
        isDisabled={false}
        lastPage={0}
        onFirstClick={[Function]}
        onLastClick={[Function]}
        onNextClick={[Function]}
        onPageInput={[Function]}
        onPreviousClick={[Function]}
        onSetPage={[Function]}
        page={0}
        pagesTitle=""
        paginationTitle="Pagination"
        perPage={20}
        toFirstPage="Go to first page"
        toLastPage="Go to last page"
        toNextPage="Go to next page"
        toPreviousPage="Go to previous page"
      >
        <nav
          aria-label="Pagination"
          className="pf-c-pagination__nav"
        >
          <div
            className="pf-c-pagination__nav-control pf-m-first"
          >
            <Button
              aria-label="Go to first page"
              data-action="first"
              isDisabled={true}
              onClick={[Function]}
              variant="plain"
            >
              <button
                aria-disabled={true}
                aria-label="Go to first page"
                className="pf-c-button pf-m-plain pf-m-disabled"
                data-action="first"
                data-ouia-component-id="OUIA-Generated-Button-plain-1"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe={true}
                disabled={true}
                onClick={[Function]}
                role={null}
                tabIndex={null}
                type="button"
              >
                <AngleDoubleLeftIcon
                  color="currentColor"
                  noVerticalAlign={false}
                  size="sm"
                >
                  <svg
                    aria-hidden={true}
                    aria-labelledby={null}
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style={
                      Object {
                        "verticalAlign": "-0.125em",
                      }
                    }
                    viewBox="0 0 448 512"
                    width="1em"
                  >
                    <path
                      d="M223.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L319.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L393.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34zm-192 34l136 136c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9L127.9 256l96.4-96.4c9.4-9.4 9.4-24.6 0-33.9L201.7 103c-9.4-9.4-24.6-9.4-33.9 0l-136 136c-9.5 9.4-9.5 24.6-.1 34z"
                    />
                  </svg>
                </AngleDoubleLeftIcon>
              </button>
            </Button>
          </div>
          <div
            className="pf-c-pagination__nav-control"
          >
            <Button
              aria-label="Go to previous page"
              data-action="previous"
              isDisabled={true}
              onClick={[Function]}
              variant="plain"
            >
              <button
                aria-disabled={true}
                aria-label="Go to previous page"
                className="pf-c-button pf-m-plain pf-m-disabled"
                data-action="previous"
                data-ouia-component-id="OUIA-Generated-Button-plain-2"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe={true}
                disabled={true}
                onClick={[Function]}
                role={null}
                tabIndex={null}
                type="button"
              >
                <AngleLeftIcon
                  color="currentColor"
                  noVerticalAlign={false}
                  size="sm"
                >
                  <svg
                    aria-hidden={true}
                    aria-labelledby={null}
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style={
                      Object {
                        "verticalAlign": "-0.125em",
                      }
                    }
                    viewBox="0 0 256 512"
                    width="1em"
                  >
                    <path
                      d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
                    />
                  </svg>
                </AngleLeftIcon>
              </button>
            </Button>
          </div>
          <div
            className="pf-c-pagination__nav-page-select"
          >
            <input
              aria-label="Current page"
              className="pf-c-form-control"
              disabled={true}
              max={0}
              min={1}
              onChange={[Function]}
              onKeyDown={[Function]}
              type="number"
              value={0}
            />
            <span
              aria-hidden="true"
            >
              of 
              0
            </span>
          </div>
          <div
            className="pf-c-pagination__nav-control"
          >
            <Button
              aria-label="Go to next page"
              data-action="next"
              isDisabled={true}
              onClick={[Function]}
              variant="plain"
            >
              <button
                aria-disabled={true}
                aria-label="Go to next page"
                className="pf-c-button pf-m-plain pf-m-disabled"
                data-action="next"
                data-ouia-component-id="OUIA-Generated-Button-plain-3"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe={true}
                disabled={true}
                onClick={[Function]}
                role={null}
                tabIndex={null}
                type="button"
              >
                <AngleRightIcon
                  color="currentColor"
                  noVerticalAlign={false}
                  size="sm"
                >
                  <svg
                    aria-hidden={true}
                    aria-labelledby={null}
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style={
                      Object {
                        "verticalAlign": "-0.125em",
                      }
                    }
                    viewBox="0 0 256 512"
                    width="1em"
                  >
                    <path
                      d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                    />
                  </svg>
                </AngleRightIcon>
              </button>
            </Button>
          </div>
          <div
            className="pf-c-pagination__nav-control pf-m-last"
          >
            <Button
              aria-label="Go to last page"
              data-action="last"
              isDisabled={true}
              onClick={[Function]}
              variant="plain"
            >
              <button
                aria-disabled={true}
                aria-label="Go to last page"
                className="pf-c-button pf-m-plain pf-m-disabled"
                data-action="last"
                data-ouia-component-id="OUIA-Generated-Button-plain-4"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe={true}
                disabled={true}
                onClick={[Function]}
                role={null}
                tabIndex={null}
                type="button"
              >
                <AngleDoubleRightIcon
                  color="currentColor"
                  noVerticalAlign={false}
                  size="sm"
                >
                  <svg
                    aria-hidden={true}
                    aria-labelledby={null}
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style={
                      Object {
                        "verticalAlign": "-0.125em",
                      }
                    }
                    viewBox="0 0 448 512"
                    width="1em"
                  >
                    <path
                      d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34zm192-34l-136-136c-9.4-9.4-24.6-9.4-33.9 0l-22.6 22.6c-9.4 9.4-9.4 24.6 0 33.9l96.4 96.4-96.4 96.4c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l136-136c9.4-9.2 9.4-24.4 0-33.8z"
                    />
                  </svg>
                </AngleDoubleRightIcon>
              </button>
            </Button>
          </div>
        </nav>
      </Navigation>
    </div>
  </Pagination>
</PaginationWrapper>
`;
